app/Console/Commands/ProductsArtistsSync.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Adapters\ShopifyAdapter;
+use App\Models\ShopifyProduct;
+use Illuminate\Console\Command;
+use Symfony\Component\Console\Helper\ProgressBar;
+
+class ProductsArtistsSync extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'migration:products:artists:sync';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Command for product-artist association';
+
+    public function __construct(protected ShopifyAdapter $shopifyAdapter)
+    {
+        parent::__construct();
+    }
+
+    /**
+     * Execute the console command.
+     */
+    public function handle()
+    {
+        $this->info("\n Creating new product-artist associations...");
+
+        $query = ShopifyProduct::query();
+
+        $progressBar = $this->output->createProgressBar($query->count());
+        $progressBar->setFormat(ProgressBar::FORMAT_VERY_VERBOSE);
+        $progressBar->start();
+
+        $query
+            ->lazyById()
+            ->each(function (ShopifyProduct $product) use ($progressBar) {
+                $migrationOriginData = $product->migrationProduct()->value('origin_data');
+
+                foreach($migrationOriginData['custom_attributes'] as $attribute) {
+                    if ('artist' === $attribute['attribute_code']) {
+                        $artistSlug = preg_replace('/\W+/', '-', $attribute['value']);
+
+                        $artist = $this->shopifyAdapter->getArtistMetaobjectBySlug($artistSlug);
```

Comment:
:large_blue_circle: I strongly recommend wrapping this one using `once()`:


```suggestion
                        $artist = once(function () use ($artistSlug) {
                            return $this->shopifyAdapter->getArtistMetaobjectBySlug($artistSlug);
                        });
```

That way, this part of code would be executed only once, and the results will be kept in memory.

For older versions of Laravel, you would have to put those in array cache, like this:

```php
cache()
    ->store('array')
    ->rememberForever('cache key', fn () => $this->shopify->getArtistMetaobjectBySlug('...'));
```

Reply:
pushed the changes